lib/pure/uri.nim

```diff
@@ -142,6 +142,69 @@ proc encodeQuery*(query: openArray[(string, string)], usePlus = true,
       result.add('=')
       result.add(encodeUrl(val, usePlus))
 
+proc decodeQuery*(urlQuery: string = ""): seq[(string, string)] =
+  ## Decode a URL query string into a set of (key, value) parameters.
+  ##
+  ## **See also:**
+  ## * `decodeUrl proc<#decodeUrl,string>`_
+  runnableExamples:
+    assert decodeQuery() == {:}
+    assert decodeQuery("a=1&b=2") == {"a": "1", "b": "2"}
+    assert decodeQuery("a=1&b") == {"a": "1", "b": ""}
+
+  var
+    name = ""
+    buffer = ""
+    encodedchar = ""
+
+  var pairs = newSeq[(string, string)]()
+
+  proc getPair(name, buffer: string): (string, string) =
+    return if name.len == 0 and buffer.len > 0: (buffer, "")
+      elif name.len > 0: (name, buffer)
+      else: ("", "")
+
+  for c in urlQuery:
+
+    if c == '&':
+      let (key, value) = getPair(name, buffer)
+      if key.len > 0:
+        pairs.add((key, value))
+      name = ""
+      buffer = ""
+      continue
+
+    if c == '=':
+      name = buffer
+      buffer = ""
+      continue
+
+    if encodedchar.len > 1:
+      encodedchar.add(c)
+      if (encodedchar.len - 1) mod 3 == 0:
+        let decodedchar = chr(fromHex[int](encodedchar))
+        if decodedchar != '\x00':
+          buffer.add(decodedchar)
+          encodedchar = ""
+          continue
+      continue
+
+    if c == '%':
+      encodedchar.add("0x")
+      continue
+
+    if c == '+':
+      buffer.add(' ')
+      continue
+
+    buffer.add(c)
+
+  let (key, value) = getPair(name, buffer)
+  if key.len > 0:
+    pairs.add((key, value))
+
+  return pairs
```

Comment:
```suggestion
  var
    name = ""
    buffer = ""
    encodedchar = ""

  proc getPair(name, buffer: string): (string, string) =
    if name.len == 0 and buffer.len > 0: (buffer, "")
    elif name.len > 0: (name, buffer)
    else: ("", "")

  for c in urlQuery:

    if c == '&':
      let (key, value) = getPair(name, buffer)
      if key.len > 0:
        result.add((key, value))
      name = ""
      buffer = ""
    elif c == '=':
      name = buffer
      buffer = ""
    elif encodedchar.len > 1:
      encodedchar.add(c)
      if (encodedchar.len - 1) mod 3 == 0:
        let decodedchar = chr(fromHex[int](encodedchar))
        if decodedchar != '\x00':
          buffer.add(decodedchar)
          encodedchar = ""
    elif c == '%':
      encodedchar.add("0x")
    elif c == '+':
      buffer.add(' ')
    else:
      buffer.add(c)

  let (key, value) = getPair(name, buffer)
  if key.len > 0:
    result.add((key, value))
```

Reply:
Removed the `continue` statements by using `elif` and used the result variable directly.